barretenberg/cpp/src/barretenberg/vm/avm/tests/merkle_tree.test.cpp

```diff
@@ -0,0 +1,130 @@
+#include "barretenberg/common/thread.hpp"
+#include "barretenberg/crypto/poseidon2/poseidon2.hpp"
+#include "barretenberg/crypto/poseidon2/poseidon2_params.hpp"
+#include "barretenberg/vm/avm/generated/circuit_builder.hpp"
+#include "barretenberg/vm/avm/generated/flavor.hpp"
+#include "barretenberg/vm/avm/generated/full_row.hpp"
+#include "barretenberg/vm/avm/generated/relations/poseidon2.hpp"
+#include "barretenberg/vm/avm/generated/relations/poseidon2_full.hpp"
+
+#include "barretenberg/vm/avm/trace/gadgets/merkle_tree.hpp"
+#include <cstdint>
+#include <cstdlib>
+#include <gtest/gtest.h>
+
+namespace tests_avm {
+using namespace bb;
+using namespace bb::Avm_vm;
+
+TEST(AvmMerkleTree, shouldCheckMembership)
+{
+
+    using FF = AvmFlavor::FF;
+    constexpr size_t TRACE_SIZE = 1 << 8;
+
+    std::vector<AvmFullRow<FF>> trace(TRACE_SIZE);
+
+    bb::avm_trace::AvmMerkleTreeTraceBuilder merkle_tree_builder;
+    std::cerr << "Generating trace of size " << TRACE_SIZE << "..." << std::endl;
+
+    /* MEMBERSHIP CHECK */
+    // Create a bunch of random elements and hash them.
+    size_t depth = 32;
+    std::vector<FF> sibling_path;
+    FF initial_value = FF::random_element();
```

Comment:
Not for now, but I think it would be good to have a proper fix test vector for a Merkle tree test. Maybe you can write a TODO on this.